### DISCUSSION TITLE
Grails 7 Scope - Should Grails 7 update to Apache Groovy 4?

### DISCUSSION THREAD

**Discussion by sdelamo** [apache/grails-core#13519] [Polls]
This is a poll to pulse the community about the Grails 7 scope. In particular about the [Apache Groovy version](https://groovy.apache.org/download.html) to be based on.

**Comment by rainboyan** [comment]
When will Groovy 5.0 be GA?

**Reply by sbglasius** [reply]
Perhaps @paulk-asert can answer that?

**Reply by paulk-asert** [reply]
We don't have a release date yet. There are a few pieces of work we wanted to include that haven't been finished yet.

**Reply by rainboyan** [reply]
@paulk-asert Thank you.

But Grails 7 is compatible with Groovy 5, which I think is possible.

**Comment by gsartori** [comment]
Grails needs to stay relevant if we want people to keep adopting it. In today’s world people are more attracted by trends than logic. Technically we could stay on Groovy 3 for years. But the Groovy websites is telling us “Groovy 3 is the previous version”. So we either remove that sentence from the Groovy website or we switch to Groovy 4. I professionally prefer to work with the “previous version” of any technology for a couple of years at least, so for what is worth in my mental framework we are ready to adopt Groovy 4.

**Comment by ppazos** [comment]
I think there are two things which Grails MUST keep up with: Groovy and Java. (and Groovy to keep up with Java too)

We need all the new features of the languages to be available right away so the framework stays relevant.

**Comment by sdelamo** [comment]
There is already a PR in progress for this https://github.com/grails/grails-core/pull/13532/files
